java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -40,54 +40,48 @@
 import org.sonar.plugins.java.api.tree.TypeCastTree;
 import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
 
-import javax.annotation.Nullable;
 import java.util.List;
 
 @Rule(key = "S3518")
 public class DivisionByZeroCheck extends SECheck {
 
-  private enum DivByZeroStatus implements ObjectConstraint.Status {
-    ZERO {
-      @Override
-      public String valueAsString() {
-        return "zero";
-      }
-    },
-    NON_ZERO {
-      @Override
-      public String valueAsString() {
-        return "non-zero";
+  private enum ZeroConstraint implements Constraint {
+    ZERO,
+    NON_ZERO,
+    UNDETERMINED;
+
+    @Override
+    public String valueAsString() {
+      switch (this) {
+        case ZERO:
+          return "zero";
+        case NON_ZERO:
+          return "non-zero";
+        case UNDETERMINED:
+          return  "\"maybe-zero\"";
       }
-    },
-    UNDETERMINED {
-      @Override
-      public String valueAsString() {
-        return "\"maybe-zero\"";
+      throw new IllegalStateException("missing case for zero constraint enum");
+    }
+
+    @Override
+    public Constraint inverse() {
+      if(this == ZERO) {
+        return NON_ZERO;
       }
+     return this;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)